src/components/NcRichText/NcReferencePicker/referencePickerModal.js

```diff
@@ -21,21 +21,22 @@ export async function getLinkWithPicker(providerId = null, isInsideViewer = unde
 			? null
 			: (getProvider(providerId) ?? null)
 
-		const View = Vue.extend(NcReferencePickerModal)
-		const view = new View({
-			propsData: {
-				initialProvider,
-				isInsideViewer,
+		const view = createApp(NcReferencePickerModal, {
+			initialProvider,
+			isInsideViewer,
+			onCancel() {
+				destroyView()
+				reject(new Error('User cancellation'))
+			},
+			onSubmit(link) {
+				destroyView()
```

Comment:
Direct access to `view` should work here without function

```suggestion
			onSubmit(link) {
				view.unmount()
```

Reply:
Indeed it does. Fixed.